### PR TITLE
get these subjects filled

### PR DESCRIPTION
**Pull request by foomatty**

### CODE REVIEW

File: topics/cfgmgmt.md [hunk 1]
```diff
@@ -2,4 +2,17 @@
...
\ No newline at end of file
+---
+
+Configuration management is paramount for building and defining reproducible services.  It is used 
+to define and create compute, network, and storage resources for provisioning of services.  Where applicable it 
+is also used to define monitoring and alerting mechanisms.
+
+### Application configuration
+
+Where applicable application and service local configuration should be abstract from the deployed code and provided
```
Comment (russellhill) [inline topics/cfgmgmt.md]: "abstracted"

File: topics/deployments.md [hunk 2]
```diff
@@ -2,4 +2,53 @@
...
----
\ No newline at end of file
+---
+
+The master branch is considered as always in a releasable state
+
+### Runbooks
+
+Each project should contain a runbook detailing :
+    * Prequisites required for deploying the service
```
Comment (russellhill) [inline topics/deployments.md]: "prerequisites"?

File: topics/deployments.md [hunk 3]
```diff
@@ -2,4 +2,53 @@
...
+
+Notification should also be sent to New Relic or a Graphite handler detailing the version released. 
+
+### Backwards compatible
+
+As deployments are not atomic, all deployments should be able to co-exist with the current live version.
+
+### Canary deployments
+
+The deployment process should include a canary staged deployments.  This can include but not limited to the following stages:
```
Comment (russellhill) [inline topics/deployments.md]: "a canary staged deployment"?

Comment (russellhill) [inline topics/deployments.md]: "can include but is not limited to"

Comment (russellhill) [inline topics/deployments.md]: do we class deployment to a staging area as a "canary deployment" - if not, then we don't follow the procedures detailed here in TN at the present time...

File: topics/monitoring.md [hunk 4]
```diff
@@ -13,12 +13,21 @@ Applications where possible should expose a status endpoint for internal monitor
...
-They can be monitored to show if they are in a running state.
+Synthetic checks should be continually executed against the application, these checks should perform common user defined functions.
+
+### Non Listening Services
 
-The result of these services should be monitored, this could be based on queues, data alteration, or the reading of log files.
\ No newline at end of file
+Example of monitoring for non listening services:
+    * Check that process is running
+    * Actively check metrics produced by the service, such as, queue length and number of running processes
```
Comment (russellhill) [inline topics/monitoring.md]: should we mention "long running queue entries" as something we'd consider worth monitoring?

**Comment by omarqureshi** [comment]
:+1:

**Comment by russellhill** [comment]
:+1: looks good!

File: topics/deployments.md [hunk 5]
```diff
@@ -2,4 +2,53 @@
...
+    * Prequisites required for deploying the service
+    * Instruction on how to deploy and rollback the service
+
+### Artifacts
+
+The format of the deployment code should either be an installable debian package, or a tar gzipped archive.
+
+### Method of deployment
+
+We currently use either Puppet or Ansible for deployments.  Depending on the type of service will depend on which method to use.
```
Comment (camillef) [inline topics/deployments.md]: "Which method to use will depend on the type of service." ?

File: topics/deployments.md [hunk 6]
```diff
@@ -2,4 +2,53 @@
...
+Ansible is more favourable for deployments that require a greater degree of orchestration.
+
+### Notifications
+
+All deployments should notify their actions in the appropriate hipchat channel.
```
Comment (camillef) [inline topics/deployments.md]: hipchat => HipChat

File: topics/deployments.md [hunk 7]
```diff
@@ -2,4 +2,53 @@
...
+
+### Canary deployments
+
+The deployment process should include a canary staged deployments.  This can include but not limited to the following stages:
+    * Deployed to production environment only accessible internally for validation
+    * Deployed to a percentage of the production environment to receive live traffic
+
+### Roll backs
+
+The deployment should be able to roll back leaving the production in its previous state, specific instructions on how to revert the production environment
```
Comment (camillef) [inline topics/deployments.md]: grammar ->
"[..] previous state. Specific instructions [..]"

File: topics/incidents.md [hunk 8]
```diff
@@ -26,4 +26,14 @@ During an incident it is imperative to keep communication channels open with reg
 3. Incident co-ordination (HipChat, Hangouts)
 4. Email updates to stakeholders
 
+### 24x7 Support Rota
+
+Talis applications and services operate 24x7x365, members of the engineering team form the 24x7 rota that is tasked with incident
```
Comment (camillef) [inline topics/incidents.md]: grammar ->  "[...] operate 24x7x365. Members of [...]"

File: topics/incidents.md [hunk 9]
```diff
@@ -26,4 +26,14 @@ During an incident it is imperative to keep communication channels open with reg
 3. Incident co-ordination (HipChat, Hangouts)
 4. Email updates to stakeholders
 
+### 24x7 Support Rota
+
+Talis applications and services operate 24x7x365, members of the engineering team form the 24x7 rota that is tasked with incident 
+managing Level 1 alerts to Talis services. 
+
+The main focus of the primary and secondary contacts is to manage the incident, this is either to resolution or on hand over to a Talis staff member.
```
Comment (camillef) [inline topics/incidents.md]: grammar -> "[...]to manage the incident, either to resolution or [...]"

File: topics/monitoring.md [hunk 10]
```diff
@@ -13,12 +13,21 @@ Applications where possible should expose a status endpoint for internal monitor
...
-### Non Listening Services
+### Real User Monitoring
+
+Apdex, transactional, and where applicable Browser monitoring should be in place to identify performance degradation.
 
-Services such as job workers and batch processes that are not externally accessible should still be monitored.
+### Auto User monitoring
 
-They can be monitored to show if they are in a running state.
+Synthetic checks should be continually executed against the application, these checks should perform common user defined functions.
```
Comment (camillef) [inline topics/monitoring.md]: grammar: either replace the comma with an 'and' or a full stop

File: topics/monitoring.md (hunk 4 above)
Comment (camillef) [inline topics/monitoring.md]: no comma after 'such as'

File: topics/deployments.md (hunk 3 above)
Comment (foomatty) [inline topics/deployments.md]: Thats right, canary deployments are only to production environments.

**Pull request merged**